pkg/kratos/service.go

```diff
@@ -0,0 +1,171 @@
+package kratos
+
+import (
+	"context"
+	"net/http"
+
+	"github.com/canonical/identity-platform-login-ui/internal/logging"
+	misc "github.com/canonical/identity-platform-login-ui/internal/misc/http"
+	"github.com/canonical/identity-platform-login-ui/internal/monitoring"
+	hClient "github.com/ory/hydra-client-go/v2"
+	kClient "github.com/ory/kratos-client-go"
+	"go.opentelemetry.io/otel/trace"
+)
+
+type Service struct {
+	kratos KratosClientInterface
+	hydra  HydraClientInterface
+
+	tracer  trace.Tracer
+	monitor monitoring.MonitorInterface
+	logger  logging.LoggerInterface
+}
+
+// We override the type from the kratos sdk, as it does not
+// get marshalled correctly into json
+type ErrorBrowserLocationChangeRequired struct {
+	Error *kClient.GenericError `json:"error,omitempty"`
+	// Points to where to redirect the user to next.
+	RedirectBrowserTo *string `json:"redirect_browser_to,omitempty"`
+}
+
+func (s *Service) CheckSession(ctx context.Context, cookies []*http.Cookie) (*kClient.Session, http.Header, error) {
+	_, span := s.tracer.Start(ctx, "kratos.FrontendApi.ToSession")
+	defer span.End()
+
+	session, resp, err := s.kratos.FrontendApi().ToSession(
+		ctx,
+	).Cookie(
+		misc.CookiesToString(cookies),
+	).Execute()
+
+	if err != nil {
+		// TODO @shipperizer we shouldn't be logging this
+		s.logger.Debugf("full HTTP response: %v", resp)
+
+		return nil, nil, err
+	}
+	return session, resp.Header, nil
+}
+
+func (s *Service) AcceptLoginRequest(ctx context.Context, identityID string, lc string) (*hClient.OAuth2RedirectTo, http.Header, error) {
```

Comment:
i see what you meant when saying that the return of the headers is a bit ugly, other way would be to apss in a `Writer` and change it inside the function....I mean not a bad thing but you'd incure in the same pain that we are having right now with testing these type of funcs https://github.com/canonical/identity-platform-login-ui/pull/73/files#diff-5e4ba31969008b305d67765d5d4c9bd50d9bcc49571f1594e97d2b152ddf1378R38-R39

all in all for now we are ok with 3 returns, worst case we can return a struct composed by the library object and the headers

```
type X struct {
    Headers http.Header
    Object interface{}
}
``` 

just to make it generic, but it looks terrible, which on a scale it's probably worse than ugly